language/move-compiler/src/to_bytecode/optimize/mod.rs

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) The Move Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::parser::ast::FunctionName;
+use move_ir_types::ast::{self as IR};
+use std::collections::{BTreeSet, HashMap};
+
+mod remove_fallthrough_jumps;
+mod remove_unnecessary_store;
+mod remove_unused_locals;
+
+pub(crate) fn code(
+    _f: &FunctionName,
+    loop_heads: &BTreeSet<IR::BlockLabel_>,
+    locals: &mut Vec<(IR::Var, IR::Type)>,
+    blocks: &mut IR::BytecodeBlocks,
+) {
+    let mut changed = true;
```

Comment:
Does this really need to be a loop like this, and on this level? Perhaps optimizations should locally decide whether they need the complexity of a fixpoint loop. Or are you expecting these are interconnected for finding a fixpoint?

Reply:
I lean slightly the other way on this point, i.e. advocating to keep the fixpoint computation here, but lose the fixpoints within each individual optimisation, because I do see the potential for, e.g. `remove_unnecessary_store` to eliminate enough code to empty a block, which `remove_empty_blocks` can then clean up.

The only reason to keep the fixpoints within each optimisations is if other optimisations' actions can interfere with opportunities for the local one, but that seems rare/unlikely.